src/main/java/appeng/parts/automation/StorageExportStrategy.java

```diff
@@ -55,26 +59,38 @@ public long transfer(StackTransferContext context, AEKey what, long amount, Acti
                 context.getActionSource(),
                 Actionable.SIMULATE);
 
+        if (extracted <= 0) {
+            return 0;
+        }
+
+        // Determine how much we're allowed to insert
+        long wasInserted;
         try (var tx = Platform.openOrJoinTx()) {
-            long wasInserted = adjacentStorage.insert(variant, extracted, tx);
+            wasInserted = adjacentStorage.insert(variant, extracted, tx);
+        }
 
-            if (wasInserted > 0) {
-                if (mode == Actionable.MODULATE) {
-                    StorageHelper.poweredExtraction(
-                            context.getEnergySource(),
-                            inv.getInventory(),
-                            what,
-                            wasInserted,
-                            context.getActionSource(),
-                            Actionable.MODULATE);
-                    tx.commit();
+        if (wasInserted > 0 && mode == Actionable.MODULATE) {
```

Comment:
This seems to always be called with `MODULATE` by the way, not sure why we have a mode parameter there in the first place.

Reply:
I opened a new issue for this, not going to break AppMek just for the sake of it.